pkg/agent/agent.go

```diff
@@ -106,10 +117,14 @@ func (k *Klocksmith) process(stop <-chan struct{}) error {
 		return err
 	}
 
-	// we are schedulable now.
-	glog.Info("Marking node as schedulable")
-	if err := k8sutil.Unschedulable(k.nc, k.node, false); err != nil {
-		return err
+	if makeSchedulable {
```

Comment:
I think this should be moved to before the `SetNodeAnnotationLabels` for `MadeUnschedulable: False`. As is, I believe if it crashes or has issues with this call, it will not restore state correctly afterwards.

Reply:
I see your point but I'm concerned with moving it above `if err := k.waitForNotOkToReboot(); err != nil {`. From the comment above that line, it seems like there might be trouble if we preemptively mark the node as schedulable before waiting for ok-to-reboot to clear. How about if I set the `MadeUnschedulable` annotation to false after the `if makeSchedulable` block?

Reply:
Ah, inded, you're right. Moving the annotation down seems like it should fix the concern I have correctly.